src/Microsoft.Azure.WebJobs.Host/Bindings/Path/BindingTemplate.cs

```diff
@@ -84,54 +87,54 @@ public static BindingTemplate FromString(string input, bool ignoreCase = false)
         /// <exception cref="InvalidOperationException">Thrown when required parameter value is not available.
         /// </exception>
-        public string Bind(IReadOnlyDictionary<string, string> parameters)
+        internal string Bind(IReadOnlyDictionary<string, object> parameters)
         {
             StringBuilder builder = new StringBuilder();
 
             if (_ignoreCase && parameters != null)
             {
                 // convert to a case insensitive dictionary
-                var caseInsensitive = new Dictionary<string, string>(parameters.Count, StringComparer.OrdinalIgnoreCase);
+                var caseInsensitive = new Dictionary<string, object>(parameters.Count, StringComparer.OrdinalIgnoreCase);
                 foreach (var pair in parameters)
                 {
                     caseInsensitive.Add(pair.Key, pair.Value);
                 }
                 parameters = caseInsensitive;
             }
 
-            foreach (BindingTemplateToken token in Tokens)
+            foreach (var token in Tokens)
             {
-                if (token.IsParameter)
-                {
-                    string value;
-                    BindingParameterResolver resolver = null;
-                    if (parameters != null && parameters.TryGetValue(token.Value, out value))
-                    {
-                        // parameter is resolved from binding data
-                        builder.Append(value);
-                    }
-                    else if (BindingParameterResolver.TryGetResolver(token.Value, out resolver))
-                    {
-                        // parameter maps to one of the built-in system binding
-                        // parameters (e.g. rand-guid, datetime, etc.)
-                        value = resolver.Resolve(token.Value);
-                        builder.Append(value);
-                    }
-                    else
-                    {
-                        throw new InvalidOperationException(String.Format(CultureInfo.CurrentCulture, "No value for named parameter '{0}'.", token.Value));
-                    }
-                }
-                else
-                {
-                    builder.Append(token.Value);
-                }
+                var value = token.Evaluate(parameters);
+                builder.Append(value);
             }
 
             return builder.ToString();
         }
 
         /// <summary>
+        /// Resolves original parameterized template into a string by replacing parameters with values provided as
+        /// a dictionary.
+        /// </summary>
+        /// <param name="parameters">Dictionary providing parameter values.</param>
+        /// <returns>Resolved string if succeeded.</returns>
+        /// <exception cref="InvalidOperationException">Thrown when required parameter value is not available.
+        /// </exception>
+        [Obsolete("Switch to the overload accepting a IReadOnlyDictionary<string, object>.")]
```

Comment:
Diff is playing some tricks here. The <string,string> is an actually an existing method. <string,obj> is the new one. 

---
In reply to: [114588153](https://github.com/Azure/azure-webjobs-sdk/pull/1117#discussion_r114588153) [](ancestors = 114588153)